pkg/scheduler/dependency_graph.go

```diff
@@ -339,65 +444,73 @@ func (sched *Scheduler) BuildDependencyGraph(options interfaces.DependencyGraphO
 		scheduledResource *ScheduledResource
 		parentContext     *GraphContext
 	}
+
+	replicas, err := sched.allocateReplicas(flow, options)
+	if err != nil {
+		return nil, err
+	}
+
 	blocks := map[string][]*Block{}
 
-	queue := list.New()
-	queue.PushFront(&Block{dependency: client.Dependency{Child: fullFlowName}})
+	for _, replica := range replicas {
+		queue := list.New()
+		queue.PushFront(&Block{dependency: client.Dependency{Child: fullFlowName}})
 
-	for e := queue.Front(); e != nil; e = e.Next() {
-		parent := e.Value.(*Block)
+		for e := queue.Front(); e != nil; e = e.Next() {
+			parent := e.Value.(*Block)
```

Comment:
This whole block looks like a nice thing to split and extract to named functions so that you can actually know what is going on based on function names.